### PR TITLE
beat logging indentation 

### PR DESCRIPTION
**Pull request by bandophahita**
From our conversation https://github.com/perrygoy/screenpy/discussions/21#discussioncomment-403009 I'm submitting this PR to add indentation to the beat output.  I opted not to include the original message when a question returns a value as the indentation alone gives the readability needed.

Here is an example of the output.
```
15:55:21  INFO [TASK] Rebecca attempts to GetToLoginPage
15:55:21  INFO    [TASK] Rebecca attempts to CheckIfLoginPageIsLoaded
15:55:21  INFO       Rebecca inspects the password Field.
15:55:21  INFO       <selenium.webdriver.remote.webelement.WebElement (session="5429de6160fa4de0584443309a0627e4", element="0.4460319907448458-1")>
15:55:21  INFO       ... hoping it's clickable
15:55:21  INFO       True
15:55:21  INFO       Rebecca inspects the username Field.
15:55:21  INFO       <selenium.webdriver.remote.webelement.WebElement (session="5429de6160fa4de0584443309a0627e4", element="0.4460319907448458-2")>
15:55:21  INFO       ... hoping it's clickable
15:55:21  INFO       True
15:55:21  INFO       Rebecca inspects the forgotpass Button.
15:55:21  INFO       <selenium.webdriver.remote.webelement.WebElement (session="5429de6160fa4de0584443309a0627e4", element="0.4460319907448458-3")>
15:55:21  INFO       ... hoping it's clickable
15:55:21  INFO       True
15:55:21  INFO Rebecca inspects the username Field.
```

### CODE REVIEW

**Review by perrygoy — APPROVED** [review]
Hey @bandophahita, thanks for submitting your PR! I'm very happy to have this functionality added to `screenpy`, it makes reading the CLI log much nicer.

I pulled down your PR and ran it locally to double-check some things. I also added some personal opinion things; if you hate them, i'm OK with you changing them back. Those three opinions are:
 * I changed 3 spaces to 4 because i live in Python most of my life now and find 3 spaces anathema.
 * I added a little `=>` before the return value.
 * I also moved the return value logging into the indentation context manager. This probably produces the biggest change to the output.
 
Here is what the log looks like now, using the extremely contrived [github search example](https://github.com/perrygoy/screenpy/tree/trunk/examples/github):

```
INFO     screenpy:pacing.py:83 ACT SEARCH
INFO     screenpy:pacing.py:109 Scene: Search For A Nonexistant Repository On Github
INFO     screenpy:pacing.py:139 Perry visits https://github.com
INFO     screenpy:pacing.py:139 Perry searches GitHub for "perrygoy/i-never-made-this-repo"
INFO     screenpy:pacing.py:139     Perry enters "perrygoy/i-never-made-this-repo" into the GitHub header's search input.
INFO     screenpy:pacing.py:155 then hits the RETURN key
INFO     screenpy:pacing.py:139     Perry waits 20 seconds using visibility_of_element_located with [search results message]
INFO     screenpy:pacing.py:139 Perry checks the results message...
INFO     screenpy:pacing.py:139     Perry reads the text from the search results message.
INFO     screenpy:pacing.py:155         => We couldn’t find any repositories matching 'perrygoy/i-never-made-this-repo'
INFO     screenpy:pacing.py:155     => We couldn’t find any repositories matching 'perrygoy/i-never-made-this-repo'
INFO     screenpy:pacing.py:139 ... hoping it's text containing "We couldn’t find any"
INFO     screenpy:pacing.py:155     => True
INFO     screenpy:pacing.py:139 Perry checks the results message...
INFO     screenpy:pacing.py:139     Perry reads the text from the search results message.
INFO     screenpy:pacing.py:155         => We couldn’t find any repositories matching 'perrygoy/i-never-made-this-repo'
INFO     screenpy:pacing.py:155     => We couldn’t find any repositories matching 'perrygoy/i-never-made-this-repo'
INFO     screenpy:pacing.py:139 ... hoping it's text containing "perrygoy/i-never-made-this-repo"
INFO     screenpy:pacing.py:155     => True
INFO     screenpy:pacing.py:139 Perry checks the number of results...
INFO     screenpy:pacing.py:139     Perry counts the number of search results items.
INFO     screenpy:pacing.py:155         => 0
INFO     screenpy:pacing.py:155     => 0
INFO     screenpy:pacing.py:139 ... hoping it's equal to 0
INFO     screenpy:pacing.py:155     => True
```

(Please ignore the doubled return values, it's because the custom questions in that example suite are extremely contrived.)

**Comment by perrygoy** [comment]
Posting the above review made me realize that `aside`s throughout the tests will not have the indent. I just pushed in a commit that fixes that, too, and now that "then hits the RETURN key" line from my approval message looks like this:

```
INFO     screenpy:pacing.py:139 Perry searches GitHub for "perrygoy/screenpy"
INFO     screenpy:pacing.py:139     Perry enters "perrygoy/screenpy" into the GitHub header's search input.
INFO     screenpy:pacing.py:156         then hits the RETURN key
```

How do you feel about that, @bandophahita? I personally think it looks all right, since that action was a continuation of the `Enter` action.

File: screenpy/pacing.py [hunk 1]
```diff
@@ -24,6 +25,43 @@
 logger = logging.getLogger("screenpy")
 
 
+class IndentManager:
+    """Handle the indentation for CLI logging."""
+
+    def __init__(self) -> None:
+        self.level = 0
+        self.indent = 4
```
[the remark this answers is not in the document]
Comment (bandophahita) [reply screenpy/pacing.py:33]: I went back and forth on the amount if indentation and realized the 'right amount' is totally subjective.  Which, for me, usually means making it possible for the user  in charge of setting how much or even if any indenting should take place. 

I haven't figured out how best to give that level of control to the user.

File: screenpy/pacing.py [hunk 2]
```diff
@@ -97,13 +135,14 @@ def wrapper(*args: Any, **kwargs: Any) -> Any:
...
             logger.info(completed_line)
             with allure.step(completed_line):
-                retval = func(*args, **kwargs)
-                if retval is not None:
-                    aside(str(retval))
+                with indent.next_level():
+                    retval = func(*args, **kwargs)
+                    if retval is not None:
+                        aside(f"{indent}=> {retval}")
+
```
[the remark this answers is not in the document]
Comment (bandophahita) [reply screenpy/pacing.py]: Oh shoot! I just realized I using f-strings here which would be a problem for anyone using python 3.5 or lower.  Is that something to worry about?  I never checked the library requirements.

File: screenpy/pacing.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (perrygoy) [reply screenpy/pacing.py:33]: I also noticed your idea for the `enabled` property. I think all of these could be handled with environment variables.

```python
self.indent = os.getenv("SCREENPY_CLI_INDENT", 4)
self.enabled = os.getenv("SCREENPY_ENABLE_INDENTATION", 1)
```

That's usually how i handle those sorts of things.

**Comment by bandophahita** [comment]
> How do you feel about that, @bandophahita? I personally think it looks all right, since that action was a continuation of the `Enter` action.

I had a similar thought.  I felt like it was a problem yet to be solved.  I was afraid that sometimes the caller would want `aside` to indent and other times they would not.  I hadn't really done more thinking about the matter beyond that.  Is there ever a time when `aside` would need to be called and not include indentation while being several layers deep in an a beat indentation?  

I can't think of any...

**Comment by perrygoy** [comment]
On the contrary, i think it would always be useful to know which actions logged the asides. That's how the functionality in the Allure report works, the asides are embedded within the actions' accordions.

File: screenpy/pacing.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (perrygoy) [reply screenpy/pacing.py]: I think we're probably fine here, as type hinting itself wasn't introduced until 3.5, so... people at <3.5 would have a lot of other problems before f-strings. :P

[the remark this answers is not in the document]
Comment (perrygoy) [reply screenpy/pacing.py]: But i do appreciate you thinking about version compatibility. I've just been aiming for the cutting edge the whole time.

File: screenpy/pacing.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (bandophahita) [reply screenpy/pacing.py:33]: I'm not especially partial to using environmental variables as package variables but in these cases I think it feels appropriate.  I do think the whole concept of package settings could be handled in it's own branch/PR.

[the remark this answers is not in the document]
Comment (perrygoy) [reply screenpy/pacing.py:33]: You're right, there's probably a more elegant way to handle this. I'm fine with leaving them hard-coded for now, and revisiting the customization at a future time.

**Comment by perrygoy** [comment]
@bandophahita, i realized that i've been following my team's procedure of waiting for the PR submitter to merge in, but maybe you're not able to do that or assumed i'd be merging in. 

If you're OK with the changes i made to your PR, feel free to merge!

**Comment by bandophahita** [comment]
I'm not seeing the option available to me.  I just assumed you would need to do the merge since you have write access to the repo.

**Comment by perrygoy** [comment]
All right, i'll do the merge—and hey look, now you've got a fancy `[Contributor]` flair!

**Pull request merged**